worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageDriverImpl.java

```diff
@@ -0,0 +1,190 @@
+package gov.cms.ab2d.worker.processor.coverage;
+
+import gov.cms.ab2d.common.model.Contract;
+import gov.cms.ab2d.common.model.CoverageMapping;
+import gov.cms.ab2d.common.model.CoveragePeriod;
+import gov.cms.ab2d.common.model.CoverageSearch;
+import gov.cms.ab2d.common.service.ContractService;
+import gov.cms.ab2d.common.service.CoverageService;
+import gov.cms.ab2d.common.service.PropertiesService;
+import gov.cms.ab2d.common.util.DateUtil;
+import gov.cms.ab2d.worker.config.CoverageUpdateConfig;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static gov.cms.ab2d.common.util.DateUtil.getAB2DEpoch;
```

Comment:
Made the change. It makes sense because the ZonedDateTime object is immutable anyway so there is no need to generate a new one each time it is called.